### ISSUE TITLE
Mapping the current directory as a plugin / theme inside the container

### ISSUE THREAD

**Issue by noplanman** [bitnami/bitnami-docker-wordpress#233]
**Description**

With the official WordPress image, I'm able to add a volume that basically maps the current directory to a plugin / theme directory within the container, like this:
```yaml
volumes:
- .:/bitnami/wordpress/wp-content/plugins/my-plugin
```

This makes it easy to set up an instance to develop my plugin / theme.
How can I accomplish this behaviour using the Bitnami WordPress image?

**Steps to reproduce the issue:**

1. Use `docker-compose.yml` from this repo.
2. Add `volumes` directive to WordPress service:
```yaml
volumes:
- .:/bitnami/wordpress/wp-content/plugins/my-plugin
```
3. Run `docker-compose up`

**Describe the results you received:**

I receive the following error on the WordPress container:
`Error executing 'postInstallation': EACCES: permission denied, rename '/opt/bitnami/wordpress/wp-content/index.php' -> '/bitnami/wordpress/wp-content/index.php'`

**Describe the results you expected:**

The container should start correctly, with the plugin / theme folder mapped to my current folder, as declared in the `volumes`.

**Version**

- Output of `docker version`:

```
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:21:31 2018
 OS/Arch:           darwin/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:29:02 2018
  OS/Arch:          linux/amd64
  Experimental:     true
```

- Output of `docker info`:

```
Containers: 6
 Running: 6
 Paused: 0
 Stopped: 0
Images: 62
Server Version: 18.06.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: error
 NodeID: 
 Error: error while loading TLS certificate in /var/lib/docker/swarm/certificates/swarm-node.crt: certificate (1 - 70nwyhmuk6llr4x01nsqwj1az) not valid after Tue, 08 Jan 2019 02:41:00 UTC, and it is currently Fri, 10 Apr 2020 20:40:56 UTC: x509: certificate has expired or is not yet valid
 Is Manager: false
 Node Address: 192.168.65.3
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.93-linuxkit-aufs
Operating System: Docker for Mac
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.786GiB
Name: linuxkit-025000000001
ID: IJ2D:BU2P:KJJY:YYVC:7ZYS:DMXL:SPMW:LGJW:BCTU:IG7B:LYJQ:DYDG
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 57
 Goroutines: 89
 System Time: 2020-04-11T12:37:14.4899406Z
 EventsListeners: 3
HTTP Proxy: gateway.docker.internal:3128
HTTPS Proxy: gateway.docker.internal:3129
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.22.0, build f46880f
docker-py version: 3.4.1
CPython version: 3.6.4
OpenSSL version: OpenSSL 1.0.2o  27 Mar 2018
```

**Comment by andresbono** [comment]
Hi @noplanman, can you check the permissions of the volume directory in the host and share it with us?

**Comment by noplanman** [comment]
When i map the whole WordPress directory, it works.
```yaml
volumes:
- wordpress_data:/bitnami/wordpress
- .:/bitnami/wordpress/wp-content/plugins/my-plugin
```
What doesn't work is just having the current directory as a volume.
```yaml
volumes:
- .:/bitnami/wordpress/wp-content/plugins/my-plugin
```

What exactly do you mean by `volume directory in the host`? You mean in the container or on my host?
On my host permissions are set to my user and 0755.
Inside the container I don't know, because the container fails to start due to the mentioned  error:
`Error executing 'postInstallation': EACCES: permission denied, rename '/opt/bitnami/wordpress/wp-content/index.php' -> '/bitnami/wordpress/wp-content/index.php'
`

**Comment by andresbono** [comment]
Hi @noplanman, thanks for the details.
Can you try to run the WordPress container with the `root` user? You can do the following in your docker-compose:

```
  wordpress:
    image: 'bitnami/wordpress:5'
    user: root
```

Please, let us know if that solves the issue.

**Comment by angusm73** [comment]
Hi, I recently ran into this issue as well.
@andresbono, your solution worked!

**Comment by andresbono** [comment]
Thanks for confirming it, @angusm73.

**Comment by noplanman** [comment]
Can confirm that it works running as root.

What is the downside of this though, if there is any?

**Comment by andresbono** [comment]
Thanks @noplanman.

Non-root container images add an extra layer of security and are generally recommended for production environments. However, because they run as a non-root user, privileged tasks are typically off-limits.

You can read more here: https://docs.bitnami.com/tutorials/work-with-non-root-containers/

**Comment by noplanman** [comment]
Ok, thanks for this!

It's for development only anyway, so works for me 👍

**Comment by marcosbc** [comment]
Hi, we have just released tag [`5.7.1-debian-10-r21`](https://github.com/bitnami/bitnami-docker-wordpress#571-debian-10-r21) of the container images which implements a lot of new features. In particular it adds support for the `WORDPRESS_PLUGINS` environment variable which you can use to install and activate a specific list of plugins.
